File: src/main/java/mediathek/gui/dialogEinstellungen/PanelPset.java
package mediathek.gui.dialogEinstellungen;

import mediathek.config.application.ApplicationConfiguration;
import mediathek.daten.DatenPset;
import mediathek.daten.ProgramSetRepository;
import mediathek.gui.dialogEinstellungen.pset.PanelPsetKurz;
import mediathek.gui.dialogEinstellungen.pset.PanelPsetLang;
import net.miginfocom.layout.AC;
import net.miginfocom.layout.CC;
import net.miginfocom.layout.LC;
import net.miginfocom.swing.MigLayout;

import javax.swing.*;
import java.awt.*;
import java.util.function.BiConsumer;

public class PanelPset extends JPanel {
    private final JFrame parentComponent;
    private final ProgramSetRepository programSets;
    private final BiConsumer<DatenPset[], String> programSetExporter;

    public PanelPset(
            JFrame parentComponent,
            ProgramSetRepository programSets,
            BiConsumer<DatenPset[], String> programSetExporter
    ) {
        this.parentComponent = parentComponent;
        this.programSets = programSets;
        this.programSetExporter = programSetExporter;

        initComponents();
        var applicationConfiguration = ApplicationConfiguration.getInstance();
        jCheckBoxAlleEinstellungen.addActionListener(_ -> {
            applicationConfiguration.setProgramSetShowAllSettings(jCheckBoxAlleEinstellungen.isSelected());
            setupPSetVisiblePanels();
        });
        jCheckBoxAlleEinstellungen.setSelected(applicationConfiguration.getProgramSetShowAllSettings());
        setupPSetVisiblePanels();
    }

    /**
     * Einstellungen zum Ansehen und Speichern der Filme anpassen.
     */
    private void setupPSetVisiblePanels() {
        jPanelPset.removeAll();
        if (jCheckBoxAlleEinstellungen.isSelected()) {
            jPanelPset.add(new PanelPsetLang(
                    parentComponent,
                    programSets,
                    programSets.getList(),
                    programSetExporter
            ), BorderLayout.CENTER);
        } else {
            jPanelPset.add(new PanelPsetKurz(parentComponent, programSets.getList()), BorderLayout.CENTER);
        }
        jPanelPset.updateUI();
    }

    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    // Generated using JFormDesigner non-commercial license
    private void initComponents() {
        jPanelPset = new JPanel();
        jCheckBoxAlleEinstellungen = new JCheckBox();

        //======== this ========
        setLayout(new MigLayout(
            new LC().insets("5").hideMode(3).gridGap("5", "5"), //NON-NLS
            // columns
            new AC()
                .grow().fill(),
            // rows
            new AC()
                .fill().gap()
                .grow().fill()));

        //======== jPanelPset ========
        {
            jPanelPset.setLayout(new BorderLayout());
        }
        add(jPanelPset, new CC().cell(0, 1));

        //---- jCheckBoxAlleEinstellungen ----
        jCheckBoxAlleEinstellungen.setText("alle Einstellungen anzeigen"); //NON-NLS
        add(jCheckBoxAlleEinstellungen, new CC().cell(0, 0));
    }// </editor-fold>//GEN-END:initComponents

    // Variables declaration - do not modify//GEN-BEGIN:variables
    // Generated using JFormDesigner non-commercial license
    private JPanel jPanelPset;
    private JCheckBox jCheckBoxAlleEinstellungen;
    // End of variables declaration//GEN-END:variables
}

File: src/main/kotlin/mediathek/gui/dialogEinstellungen/PanelPset.kt
/*
 * Copyright (c) 2026 derreisende77.
 * This code was developed as part of the MediathekView project https://github.com/mediathekview/MediathekView
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package mediathek.gui.dialogEinstellungen

import mediathek.config.application.ApplicationConfiguration
import mediathek.daten.DatenPset
import mediathek.daten.ProgramSetRepository
import mediathek.gui.dialogEinstellungen.pset.PanelPsetKurz
import mediathek.gui.dialogEinstellungen.pset.PanelPsetLang
import mediathek.tool.ReplacementRules
import java.awt.BorderLayout
import java.util.function.BiConsumer
import javax.swing.JFrame

class PanelPset(
    private val parentComponent: JFrame?,
    private val programSets: ProgramSetRepository,
    private val replacementRules: ReplacementRules,
    private val programSetExporter: BiConsumer<Array<DatenPset>, String>,
) : PanelPsetBase() {
    init {
        configureComponentMetadata()

        val config = ApplicationConfiguration.getInstance()
        jCheckBoxAlleEinstellungen.addActionListener {
            config.programSetShowAllSettings = jCheckBoxAlleEinstellungen.isSelected
            showSelectedProgramSetPanel()
        }
        jCheckBoxAlleEinstellungen.isSelected = config.programSetShowAllSettings
        showSelectedProgramSetPanel()
    }

    private fun configureComponentMetadata() {
        jCheckBoxAlleEinstellungen.name = PanelPsetComponentNames.SHOW_ALL_SETTINGS
        jPanelPset.name = PanelPsetComponentNames.SETTINGS_CONTENT
    }

    /**
     * Einstellungen zum Ansehen und Speichern der Filme anpassen.
     */
    private fun showSelectedProgramSetPanel() {
        jPanelPset.removeAll()
        val settingsPanel = if (jCheckBoxAlleEinstellungen.isSelected) {
            PanelPsetLang(
                parentComponent,
                programSets,
                programSets.list,
                programSetExporter,
                replacementRules,
            )
        } else {
            PanelPsetKurz(parentComponent, programSets.list)
        }
        jPanelPset.add(settingsPanel, BorderLayout.CENTER)
        jPanelPset.updateUI()
    }
}

internal object PanelPsetComponentNames {
    const val SHOW_ALL_SETTINGS = "PanelPset.showAllSettings"
    const val SETTINGS_CONTENT = "PanelPset.settingsContent"
}
